src/coreclr/jit/optimizer.cpp

```diff
@@ -6792,6 +6809,9 @@ void Compiler::optHoistThisLoop(unsigned lnum, LoopHoistContext* hoistCtxt, Basi
     defExec.Push(pLoopDsc->lpEntry);
 
     optHoistLoopBlocks(lnum, &defExec, hoistCtxt);
+
+    const unsigned numHoisted = pLoopDsc->lpHoistedFPExprCount + pLoopDsc->lpHoistedExprCount;
+    return pLoopDsc->lpHoistAddedPreheader || (numHoisted > 0);
```

Comment:
Sorry for getting back late, but what is the scenario where `lpHoistAddedPreheader == true` but `numHoisted == 0`?

Reply:
Yes, there were a few cases in SPMI where we created a preheader and then failed the eh check.

Since we've modified the flow graph (even though we might not have hoisted anything) we need to return true.